fuel-asm/src/encoding_tests.rs

```diff
@@ -54,6 +53,29 @@ fn opcode() {
     }
 }
 
+#[test]
+fn invalid_reserved_part() {
+    // Args: 0
+    Instruction::try_from([Opcode::NOOP as u8, 0, 0, 0]).unwrap();
+    Instruction::try_from([Opcode::NOOP as u8, 0, 0, 1])
+        .expect_err("Reserved part is nonzero");
+
+    // Args: 1
+    Instruction::try_from([Opcode::RET as u8, 0, 0, 0]).unwrap();
+    Instruction::try_from([Opcode::RET as u8, 0, 0, 1])
+        .expect_err("Reserved part is nonzero");
+
+    // Args: 2
+    Instruction::try_from([Opcode::NOT as u8, 0, 0, 0]).unwrap();
+    Instruction::try_from([Opcode::NOT as u8, 0, 0, 1])
+        .expect_err("Reserved part is nonzero");
+
+    // Args: 3
+    Instruction::try_from([Opcode::XOR as u8, 0, 0, 0]).unwrap();
+    Instruction::try_from([Opcode::XOR as u8, 0, 0, 1])
+        .expect_err("Reserved part is nonzero");
+}
```

Comment:
While this behaviour is implemented as a macro, and all opcodes inherit this behaviour, it may be worth testing this for all opcodes. From an opcode-centric perspective, we want to show that each opcode fulfills the specification (regardless of implementation). 

Maybe we could have a macro-generated test(s) that tests common behaviours for a given opcode, and maintain a list that invokes this test macro for all opcodes.

But that's up to you - it may be overkill here.

Reply:
Makes sense to me.